### ISSUE TITLE
pyright gets confused by `from .foo import foo`

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#3131]
Create `foo.py` with:

```python
def foo(): pass
```

Create `__init__.py` with:

```python
from .foo import foo

foo()

def what():
    foo()
```

Run `pyright __init__.py` and get:
```
__init__.py:6:5 - error: Module is not callable (reportGeneralTypeIssues)
```

**Comment by not-my-profile** [comment]
Oh thanks, I didn't know that `from .foo import bar` brings `foo` into scope. Apparently neither do mypy and pytype.

**Comment by judfs** [comment]
I'm not seeing the exact issue the op described. Recreating the example as `my_package`, running `my_package/__init__.py` produces no errors.

however if I make `run2.py`
```
import my_package.foo
my_package.foo()
my_package.foo.foo()
```

I get 
```
.../bug/run2.py:2:1 - error: Module is not callable (reportGeneralTypeIssues)
```

But worse imo, if I actually run it I get a runtime error pyright didn't detect:
```
Traceback (most recent call last):
  File "run2.py", line 3, in <module>
    my_package.foo.foo()
AttributeError: 'function' object has no attribute 'foo'
```


Tested with pyright 1.1.339

**Comment by debonte** [comment]
@judfs, this issue has been closed for a long time. Please open a new issue.
